pkg/applications/github/on_tag_created.go

```diff
@@ -0,0 +1,193 @@
+package github
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"slices"
+
+	"github.com/mitchellh/mapstructure"
+	"github.com/superplanehq/superplane/pkg/configuration"
+	"github.com/superplanehq/superplane/pkg/core"
+)
+
+type OnTagCreated struct{}
+
+type OnTagCreatedMetadata struct {
+	Repository *Repository `json:"repository"`
+}
+
+type OnTagCreatedConfiguration struct {
+	Repository string                    `json:"repository"`
+	Tags       []configuration.Predicate `json:"tags"`
+}
+
+func (t *OnTagCreated) Name() string {
+	return "github.onTagCreated"
+}
+
+func (t *OnTagCreated) Label() string {
+	return "On Tag Created"
+}
+
+func (t *OnTagCreated) Description() string {
+	return "Listen to GitHub tag creation events"
+}
+
+func (t *OnTagCreated) Icon() string {
+	return "github"
+}
+
+func (t *OnTagCreated) Color() string {
+	return "gray"
+}
+
+func (t *OnTagCreated) Configuration() []configuration.Field {
+	return []configuration.Field{
+		{
+			Name:     "repository",
+			Label:    "Repository",
+			Type:     configuration.FieldTypeString,
+			Required: true,
+		},
+		{
+			Name:     "tags",
+			Label:    "Tags",
+			Type:     configuration.FieldTypeAnyPredicateList,
+			Required: true,
+			Default: []map[string]any{
+				{
+					"type":  configuration.PredicateTypeMatches,
+					"value": ".*",
+				},
+			},
+			TypeOptions: &configuration.TypeOptions{
+				AnyPredicateList: &configuration.AnyPredicateListTypeOptions{
+					Operators: configuration.AllPredicateOperators,
+				},
+			},
+		},
+	}
+}
+
+func (t *OnTagCreated) Setup(ctx core.TriggerContext) error {
+	var metadata OnTagCreatedMetadata
```

Comment:
We also have a common function to verify the repository and save it on the metadata. Check ensureRepoInMetadata in common.go